substrate/pkg/controller/substrate/cluster/instance.go

```diff
@@ -133,3 +131,15 @@ func (i *Instance) delete(ctx context.Context, substrate *v1alpha1.Substrate, pr
 	logging.FromContext(ctx).Infof("Deleted instances %v", aws.StringValueSlice(instances))
 	return nil
 }
+
+func tagsMatch(ec2Tags []*ec2.Tag, substrate *v1alpha1.Substrate) bool {
+	for _, tag := range ec2Tags {
+		if aws.StringValue(tag.Key) == "kit.aws/subtrate" && aws.StringValue(tag.Value) != substrate.Name {
```

Comment:
isn't this already filtering instances based on the tag you're looking for -  [here](https://github.com/awslabs/kubernetes-iteration-toolkit/pull/376/files#diff-02ec6b860e5c6ddc13f877ae4bec3be0234e51ac83c05dad0c98dbc765afaab9R38)  with help of this [function](https://github.com/awslabs/kubernetes-iteration-toolkit/blob/17c7b32fa3b383484d216590146aa6b7c0cfdabb/substrate/pkg/utils/discovery/tags.go#L40-L49) ?  Why do we want to re-check again ?